tests/manage/z_cluster/cluster_expansion/test_add_capacity_with_node_restart.py

```diff
@@ -0,0 +1,202 @@
+    argvalues=[
+        pytest.param(
+            *['worker', 1],
+            marks=pytest.mark.polarion_id("OCS-1311")
+        ),
+     ]
+)
+@pytest.mark.parametrize(
+    "workload_storageutilization_rbd",
+    [(0.02, False, 120)],
+    indirect=["workload_storageutilization_rbd"]
+)
+# @pytest.mark.parametrize(
+#     "workload_storageutilization_cephfs",
+#     [(0.05, False, 120)],
+#     indirect=["workload_storageutilization_cephfs"]
+# )
+@ignore_leftovers
+@tier4
+@polarion_id('OCS-1313')
+class TestAddCapacityNodeRestart(ManageTest):
+    """
+    Test add capacity when one of the nodes got restart
+    in the middle of the process. Don't forget to include the flag '--cluster-name' when running the test
+    """
+    num_of_pvcs = 4
+    pvc_size = 100
+
+    def test_add_capacity_node_restart(
+        self, nodes, multi_pvc_factory, pvc_factory, pod_factory, workload_storageutilization_rbd,
+        node_type, num_of_nodes,
+    ):
+        """ test add capacity when one of the nodes got restart
+        in the middle of the process
+        """
+        logging.info("Condition 1 to start the test is met: storageutilization is completed")
+        used_capacity_after_utilizing = temp_helper_file.get_percent_used_capacity()
+        logging.info(f"### used capacity after storageutilization = {used_capacity_after_utilizing}")
+
+        seconds_to_wait_after_storageutilization = 60
+        logging.info(f"wait {seconds_to_wait_after_storageutilization} seconds before starting the test")
+        time.sleep(seconds_to_wait_after_storageutilization)
+
+        assert temp_helper_file.check_pods_in_running_state(), \
+            "Condition 2 to start test failed: one or more OCS pods are not in running state"
+
+        node_list = get_typed_nodes(node_type=node_type, num_of_nodes=num_of_nodes)
+        assert node_list, "Condition 3 to start test failed: No node to restart"
+
+        osd_pods_before = pod_helpers.get_osd_pods()
+        assert len(osd_pods_before) < 9, \
+            "Condition 4 to start test failed: We have maximum of osd's in the cluster"
+        logging.info("All start conditions are met!")
+
+        total_space_b4_expansion = get_total_space()
+        logging.info('total space  before expansion = {}'.format(str(total_space_b4_expansion)))
+
+        logging.info("Perform some IO operations...")
+
+        pvc_objs = multi_pvc_factory(
+            size=self.pvc_size, num_of_pvc=self.num_of_pvcs
+        )
+
+        pod_objs = []
+
+        with ThreadPoolExecutor(max_workers=self.num_of_pvcs) as executor:
+            for pvc_obj in pvc_objs:
+                executor.submit(pod_objs.append, (pod_factory(pvc=pvc_obj)))
+
+        with ThreadPoolExecutor(max_workers=self.num_of_pvcs-1) as executor:
+            for pod_obj in pod_objs:
+                pod_io_task = executor.submit(
+                    pod_obj.run_io, storage_type='fs', size=4, jobs=10, io_direction='wo', rate='250m')
+
```

Comment:
Need to cover NooBaa IOs also....

Reply:
@hnallurv maybe in a follow up PR? this one is already quite large

Reply:
The IOs here are not complete. When the branch 'wip-add-capacity-e_e' will be merged into master I will use the functions from this branch. Currently, I think there is no need to reinvent the wheel and we don't want to wait for this PR until branch 'wip-add-capacity-e_e' will be merged into master, cause it may take some time.